lib/plugins/audit.js

```diff
@@ -0,0 +1,184 @@
+ /**
+ * Copyright (c) 2017-present, Jembi Health Systems NPC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+'use strict'
+
+const logger = require('winston')
+const moment = require('moment')
+
+const EVENT_OUTCOME = {
+  SUCCESS: 0,
+  MINOR_FAILURE: 4,
+  SERIOUS_FAILURE: 8,
+  MAJOR_FAILURE: 12
+}
+
+const excludedResourceTypeInteractions = [
+  {
+    type: 'AuditEvent',
+    interactions: '*'
+  },
+  {
+    type: 'user',
+    interactions: 'authenticate' // AuditEvent expects an authenticated request which will fail here - Audit wont be logged
+  }
+]
+
+module.exports = (mongo, fhirResources) => {
+  const fhirCore = require('../fhir/core.js')(mongo, fhirResources)
+
+  const getSuccessOrFailed = (data) => {
+    if (data && data.resourceType === 'OperationOutcome' && data.issue) {
+      return EVENT_OUTCOME.MINOR_FAILURE
+    }
+    return EVENT_OUTCOME.SUCCESS
+  }
+
+  const buildEventObj = (interaction, data) => {
+    return {
+      type: {
+        coding: [
+          {
+            system: 'http://hl7.org/fhir/audit-event-type',
+            code: 'rest',
+            display: 'Restful Operation'
+          }
+        ]
+      },
+      subtype: [
+        {
+          coding: [
```

Comment:
This coding array isn't correct according to the spec and needs to be removed. system and code should be in the type property directly.